core/src/main/scala/org/apache/spark/util/collection/ExternalSorter.scala

```diff
@@ -400,13 +395,12 @@ private[spark] class ExternalSorter[K, V, C](
       mergeCombiners: (C, C) => C,
       comparator: Comparator[K],
       totalOrder: Boolean)
-      : Iterator[Product2[K, C]] =
-  {
+      : Iterator[Product2[K, C]] = {
     if (!totalOrder) {
       // We only have a partial ordering, e.g. comparing the keys by hash code, which means that
       // multiple distinct keys might be treated as equal by the ordering. To deal with this, we
       // need to read all keys considered equal by the ordering at once and compare them.
-      new Iterator[Iterator[Product2[K, C]]] {
+      val it = new Iterator[Iterator[Product2[K, C]]] {
```

Comment:
Nit (more like a question): why a new `val` is introduced here?

Reply:
I was trying to convert the `flatMap(i => i)` below to simply `flatten`. For reasons even I'm not clear about, I had to introduce an intermediate val here then return `it.flatten` to get it to work. Seems harmless enough as a change but I didn't get the difference in type inference